app/models/tag.rb

```diff
@@ -323,27 +325,27 @@ def cytoscape_edge(related_tag)
 
   # published sections are sections that belong to a published lecture
   def visible_sections(user)
-    user.filter_sections(sections).select { |s|
+    user.filter_sections(sections).select do |s|
       s.lecture.visible_for_user?(user)
-    }
+    end
   end
 
   def cache_key
-    super + '-' + I18n.locale.to_s
+    "#{super}-#{I18n.locale}"
   end
 
   def touch_lectures
-    Lecture.where(id: sections.map(&:lecture)
-                              .map(&:id)).update_all updated_at: Time.now
+    Lecture.where(id: sections.map { |section| section.lecture.id })
+           .update(updated_at: Time.current)
   end
```

Comment:
All changes from `update_all` to `update` are extremely problematic and introduce unwanted behaviour. The reason is that `update` triggers Active Record Callbacks whereas `update_all` does not, see [here](https://apidock.com/rails/ActiveRecord/Relation/update_all). If you use `update` here and in the other places where it replaced `update_all`, you will get infinite update loops. Just try `/tags/5/edit`
and add an alias to the tag. This loop is easily explained: The `touch_lectures` method now triggers updates for the associated lectures, which (among others) triggers the `touch_lessons` after save callback on the `lecture` model which then triggers the `touch_media` after save callback on the lesson model, and so on and so on. The `update_all` was placed deliberately in the places where it was before to prevent such cascades. There seems to be a `touch_all` method that [does exactly what we want](https://api.rubyonrails.org/v7.0.4/classes/ActiveRecord/Relation.html#method-i-touch_all). I would suggest we use this in all instances where we had an `update_all(updated_at: Time.now)` before and add `touch_all` to the list of allowed methods on Rails/SkipsModelValidations. Beforehand, we have to undo exactly these replacements of `update_all` to `update`. Is there a simple way to do this?

Reply:
Good to know that this is so problematic. Here is what I've done to fix this in commit 461697476d5:

- In the diffs, search for "Rails/SkipsModelValidations": `git log -p -S "Rails/SkipsModelValidations"`
- Check the lines manually, go to the respective file and replace the method call (if the method call included the word `Time` and `updated_at:`)
- In VSCode search: manually check occurrences of "Time.now" and "Time.current" and "Time.zone.now" to make sure they do not occur in an `update` method call (in migrations it's ok). With this approach I changed a few more occurences: `comments_controller.rb`, `medium_controller.rb`, `answer.rb`, in `user.rb` we have `consented_at: Time.zone.now` which I left as is)

Reply:
I commented on some lines in [https://github.com/MaMpf-HD/mampf/commit/461697476d51a8b23f05d879b533b048ed6ed7c5](https://github.com/MaMpf-HD/mampf/commit/461697476d51a8b23f05d879b533b048ed6ed7c5), for some reason they does not show up in the main dialogue.

Reply:
Fixed it in 58874f70b. I also replaced `touch_all` by `touch` in the media controller where you noted there is a chance we actually need to call the callbacks. Well, let's hope it's not the case. If it is, we know how to quickly solve it and can then write a comment over there such that this is no longer implicit knowledge.